files/class-vip-filesystem.php

```diff
@@ -385,4 +410,77 @@ public function filter_get_attached_file( $file, $attachment_id ) {
 
 		return $file;
 	}
+
+	/**
+	 * Filter `cron_schedules` output
+	 *
+	 * Add a custom schedule for a 5 minute interval
+	 *
+	 * @param   array   $schedule
+	 *
+	 * @return  mixed
+	 */
+	public function filter_cron_schedules( $schedule ) {
+		$schedule['five_minutes'] = [
+			'interval' => 300,
+			'display' => __( 'Once every 5 minutes' ),
+		];
+
+		return $schedule;
+	}
+
+	public function schedule_update_job() {
+		if ( get_option( self::OPT_ALL_FILESIZE_PROCESSED ) ) {
+			wp_clear_scheduled_hook( self::CRON_EVENT_NAME );
+
+			return;
+		}
+
+		if (! wp_next_scheduled ( self::CRON_EVENT_NAME )) {
+			wp_schedule_event(time(), 'five_minutes', self::CRON_EVENT_NAME );
+		}
+
+		add_action( self::CRON_EVENT_NAME, [ $this, 'update_attachment_meta' ] );
+	}
+
+	/**
+	 * Cron job to update attachment metadata with file size
+	 */
+	public function update_attachment_meta() {
+		trigger_error(
+			sprintf( 'Starting %s... $vip-go-streams-debug', self::CRON_EVENT_NAME ),
+			E_USER_NOTICE );
+		if ( get_option( self::OPT_ALL_FILESIZE_PROCESSED ) ) {
+			// already done. Nothing to update
+			trigger_error(
+				sprintf( 'Already completed. Exiting %s... $vip-go-streams-debug', self::CRON_EVENT_NAME ),
+				E_USER_NOTICE );
+			return;
+		}
+
+		$updater = new Meta_Updater( 1000 );
+		$max_id = $updater->get_max_id();
+		$start_index = get_option( self::OPT_NEXT_FILESIZE_INDEX, 0 );
+		$end_index = $start_index + $updater->get_batch_size();
+
+		if ( $start_index > $max_id ) {
+			// This means all attachments have been processed so marking as done
+			update_option( self::OPT_ALL_FILESIZE_PROCESSED, 1 );
+
+			return;
+		}
+
+		$attachments = $updater->get_attachments( $start_index, $end_index );
```

Comment:
One thought I had: on some sites, there will be batches where there are no attachments, in which case, we're just wasting time waiting for the next cron run. I wonder if we should move this to a `while` loop and run until we get some returned?